cmd/server/bookmark/repository.go

```diff
@@ -0,0 +1,63 @@
+package bookmark
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+)
+
+type Repository struct {
+	DB *sql.DB
+}
+
+func NewRepository(db *sql.DB) *Repository {
+	return &Repository{DB: db}
+}
+
+// Transactionally executes a function within a database transaction. It commits the transaction
+// if the function succeeds, otherwise it rolls back. If rollback fails, both errors are returned.
+func (r *Repository) Transactionally(ctx context.Context, f func(tx *sql.Tx) error) error {
+	tx, err := r.DB.BeginTx(ctx, nil)
+	if err != nil {
+		return err
+	}
+
+	err = f(tx)
```

Comment:
We should also catch any panics from f(tx). Could be a todo for now.

Reply:
Done ✅